src/org/authInfo.ts

```diff
@@ -344,32 +359,50 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
         .map((c) => c.key);
       try {
         const authInfo = await AuthInfo.create({ username });
-        const { orgId, instanceUrl, devHubUsername } = authInfo.getFields();
+        const { orgId, instanceUrl, devHubUsername, expirationDate, isDevHub } = authInfo.getFields();
         final.push({
           aliases,
           configs,
           username,
           instanceUrl,
-          isScratchOrg: Boolean(devHubUsername),
+          isScratchOrg: Boolean(devHubUsername) ?? false,
+          isDevHub: isDevHub || false,
           orgId: orgId as string,
           accessToken: authInfo.getConnectionOptions().accessToken,
           oauthMethod: authInfo.isJwt() ? 'jwt' : authInfo.isOauth() ? 'web' : 'token',
+          isExpired:
+            Boolean(devHubUsername) && expirationDate
+              ? new Date(ensureString(expirationDate)).getTime() < new Date().getTime()
+              : false,
         });
       } catch (err) {
         final.push({
           aliases,
           configs,
           username,
-          orgId: auth.orgId,
-          instanceUrl: auth.instanceUrl,
+          orgId: org.orgId,
+          instanceUrl: org.instanceUrl,
           accessToken: undefined,
           oauthMethod: 'unknown',
           error: err.message,
+          isExpired: false,
```

Comment:
Is it accurate to say that these orgs aren't expired? Maybe `isExpired` should be `boolean | 'unknown'`?

Reply:
Since the only orgs where is expired is well-known is scratch orgs, I thought that defaulting to false was simplest thing to do, but I can see unknown as a alternative to a default of false.